website/server/libs/spells.js

```diff
@@ -163,14 +164,35 @@ async function castSpell (req, res, { isV3 = false }) {
       task: results[1],
     });
   } else if (targetType === 'self') {
-    await castSelfSpell(req, user, spell, quantity);
+    // Check if chilling frost skill has been previously casted or not.
+    // See #12361 for more details.
+    const spellName = spell.key;
+    if (!chillingFrostAlreadyCast(spellName, user)) {
```

Comment:
sorry, @JalanshMunshi is there a reason why the check is made here and not in common code so that it applies to the client side as well?

Reply:
@paglias Would you be so kind as to please clarify where should I add it in common code? I am not able to understand what you're trying to say because the function has been implemented in `common` code and I am calling that function in server code.

Also, it seems that there is something wrong with this as the sanity check says that this import will not work in production. Can you please help me out?

Reply:
The sanity check is because importing common modules cannot be done as a normal import, but in this case what I meant was to move the check here https://github.com/HabitRPG/habitica/blob/develop/website/common/script/content/spells.js#L107 so that the code is executed both on the client and the server.

You can see an example of something similar being done for the healing spell here https://github.com/HabitRPG/habitica/blob/develop/website/common/script/content/spells.js#L230

Does it make sense?

Reply:
@paglias Yes, that helps! Works as expected.

Btw, I did what @SabreCat suggested.
> For that we'd need to e.g. change line 45 of website/client/src/components/tasks/spells.vue to @click="!spellDisabled(key) ? castStart(skill) : null"

The button becomes disabled, as the code clearly suggests. But the API is never called (I put a `console.log` for `Stealth` skill and checked). As per the issue description, the API should display an error. This change failed the test cases that I wrote for the new changes. 

We can avoid the deduction by keeping the check here - 
https://github.com/HabitRPG/habitica/blob/a501b2a5a631bc06f6c45c2bd023f15a51913094/website/common/script/content/spells.js#L713-L716

But again, we will be able to do this for `Chilling Frost`, but not for `Stealth` as it requires an import of `Tasks` in common which is not possible as per the previous discussion in this PR.

What do you suggest we do in this case?

Reply:
> But the API is never called (I put a console.log for Stealth skill and checked). As per the issue description, the API should display an error. This change failed the test cases that I wrote for the new changes.

This is expected as the API call is made when the button is clicked, the issue is about the API retuning an error in this case but if it's not called it cannot return anything, but it's fine the important thing is that the moment the API is called for example with a script or from the mobile apps it return an error

Reply:
I'm sorry but I'm a bit confused here:

### For stealth
- the check requires access to the tasks meaning that there are two different checks, one on the client (already added in a previous pr) and one on the server side added in this PR

### For chilling frost
- since the check doesn't require access to the tasks it can be done in a single place inside the common code that gets run both on the client and on the server and it's called on the client before the MPs are deducted if you do as I suggested above and move the check in here https://github.com/HabitRPG/habitica/blob/5bd4c6af4116186e51419a2f1721dcebb9fac606/website/common/script/content/spells.js#L107 instead of calling it in this file that is server side only

Let me know if something isn't clear

Reply:
Yup, that solution for the client will mean there's no API call made--which is a good thing. We could force a snackbar notification, but with the tooltip saying "there's no need to cast this" I think it's OK to simply have a null response to the click event.

For manual testing of your API changes using the site, you'll need to have the old `@click` code in place!

Reply:
Ah, I understand now. So sorry for the confusion. Yes, we will indeed need the changes in the API in case it is called by a script or mobile app. 

I have made the requested change in the client code (in `spells.vue`). The appropriate changes to the test case have also been done. They are passing when I use the old code of `spells.vue`. Please check from your end as well. 

Lastly, I started working on the wording change and we will need to change all the JSON files for different languages. I have added the changes for three files in my last commit. Could you please check it and let me know if it is okay? I will modify the other files accordingly. 

Let me know if anything needs to be changed elsewhere! :smiley:

Reply:
Only `website/common/locales/en` needs to be changed for language JSON. The rest is handled via Weblate, translate.habitica.com!

What test case is failing when you change the `spells.vue` line? I don't see `client-unit` failing at all.

Reply:
@SabreCat Oh, I didn't know that. That's awesome! I will revert the unnecessary changes then.

I am talking about the API-V3 Integrations tests. I added 2 new test cases in V3 Integration tests. Am I missing something here?

Reply:
Nope, that makes sense. I was puzzled because a change in a `.vue` file should only affect client tests, if anything, but it sounded like you were saying that the `spells.vue` change caused a server test failure!